### ISSUE TITLE
Asymmetric fee snapshot in accrue_fees_inner overpays junior tranche during active insurance loss

### ISSUE THREAD

**Issue by v1ktorrr0x** [dcccrypto/percolator-stake#182]
## Summary

In `accrue_fees_inner`, the tranche fee snapshot uses an **asymmetric pair** of balances: junior is snapshotted at `pool.junior_balance()` (gross, pre-loss) while senior is derived from `pool.senior_balance()` which equals `total_pool_value() - effective_junior_balance()` (loss-adjusted). This causes junior to systematically receive a disproportionate fee share whenever a live insurance loss is outstanding.

## Code

```rust
// src/processor.rs — accrue_fees_inner
let (snapshot_junior_bal, snapshot_senior_bal) = if distribute_to_junior {
    (
        pool.junior_balance(),                               // ← gross, NOT loss-adjusted
        pool.senior_balance().ok_or(StakeError::Overflow)?,  // ← loss-adjusted
    )
} else {
    (0, 0)
};
```

And `senior_balance()` in `state.rs`:

```rust
pub fn senior_balance(&self) -> Option<u64> {
    self.total_pool_value()?
        .checked_sub(self.effective_junior_balance()) // loss-adjusted
}
```

## Impact

During an active insurance loss (`net_loss = total_flushed - total_returned > 0`):

- `junior_balance()` = gross stored value (e.g. 1000)
- `effective_junior_balance()` = 1000 − junior_loss_share (e.g. 700)
- `senior_balance()` = `total_pool_value()` − 700

`distribute_fees` receives junior weight = 1000 (inflated) while senior weight is derived from the loss-adjusted pool. Junior earns a larger fee share than its current collateral value warrants — **at the exact moment it is supposed to be absorbing first-loss exposure**. Every `AccrueFees` call during the loss window transfers fee income from senior LP holders to junior LP holders.

## Severity

**Medium** — requires tranches enabled + active insurance loss + trading pool (mode 1). All conditions reachable simultaneously in normal operation.

## Fix

Replace `pool.junior_balance()` with `pool.effective_junior_balance()` in the snapshot so both sides use the same loss-adjusted basis:

```rust
let snapshot_junior_bal = pool.effective_junior_balance();
let snapshot_senior_bal = pool.senior_balance().ok_or(StakeError::Overflow)?;
```

**Comment by dcccrypto** [comment]
Verified NOT_REAL against current code: the suggested effective_junior_balance() swap is a no-op on all reachable paths (net_loss is 0 on mode-1 pools — tranche-loss and trading-fee accrual are mutually exclusive states). No fee asymmetry exists. Closing.
